python/Ganga/Utility/Runtime.py

```diff
@@ -257,12 +257,14 @@ def loadPlugins(environment):
             logger.error('problems with loading Named Templates for %s', n)
             logger.error('Reason: %s' % str(err))
 
-    for r in allRuntimes.values():
+    for n, r in zip(allRuntimes.keys(), allRuntimes.values()):
         try:
             r.loadPlugins()
         except Exception as err:
-            logger.error('problems with loading Named Templates for %s', n)
+            from Ganga.Core.exceptions import GangaException
+            logger.error('problems with loading Plugin %s', n)
             logger.error('Reason: %s' % str(err))
+            raise GangaException("Failed to load plugin: %s. Ganga will now shutdown to prevent job corruption." % n)
```

Comment:
1. I'm reluctant to start going around making new exceptions for all the different situations, we just end up with a dictionary of possible classes in the exception which doesn't really shed much light imho.
 2. Fine, I'll try a global import again, but once again I strongly feel this is the wrong thing to be doing here as we shouldn't really have Core classes being imported into Utility imo as well as the amount of extra stuff pulled in out of a sane order or safe way when we do this.